helm/resource_release.go

```diff
@@ -797,18 +842,49 @@ func getValues(d resourceGetter) (map[string]interface{}, error) {
 		}
 	}
 
-	yaml, err := yaml.Marshal(base)
-	if err == nil {
-		yamlString := string(yaml)
-		for _, raw := range d.Get("set_sensitive").(*schema.Set).List() {
-			set := raw.(map[string]interface{})
-			yamlString = strings.Replace(yamlString, set["value"].(string), "<SENSITIVE>", -1)
+	return base, logValues(base, d)
+}
+
+func getValue(base, set map[string]interface{}) error {
+	name := set["name"].(string)
+	value := set["value"].(string)
+	typ := set["type"].(string)
+
+	switch typ {
+	case "auto", "":
+		if err := strvals.ParseInto(fmt.Sprintf("%s=%s", name, value), base); err != nil {
+			return fmt.Errorf("failed parsing key %q with value %s, %s", name, value, err)
+		}
+	case "string":
+		if err := strvals.ParseIntoString(fmt.Sprintf("%s=%s", name, value), base); err != nil {
+			return fmt.Errorf("failed parsing key %q with value %s, %s", name, value, err)
 		}
+	default:
+		return fmt.Errorf("unexpected type: %s", typ)
+	}
 
-		log.Printf("---[ values.yaml ]-----------------------------------\n%s\n", yamlString)
+	return nil
+}
+
+func logValues(values map[string]interface{}, d resourceGetter) error {
+	// copy array to avoid change values by the cloak function.
+	asJSON, _ := json.Marshal(values)
```

Comment:
Should handle this err and return it?

Reply:
I can't imagine how this can fail since there is not any custom marshaller.

Reply:
Yeah you're right. The worst thing that could happen here is someone uses the function to try and log a map with types that couldn't be marshalled, at which point it would just log `null`